apps/ehr/src/features/css-module/components/medication-administration/InteractionAlertsDialog.tsx

```diff
@@ -0,0 +1,307 @@
+  onCancel: () => void;
+  onContinue: (interactions: MedicationInteractions) => void;
+}
+
+const SEVERITY_ORDER = ['high', 'moderate', 'low'];
+
+const SEVERITY_TO_LABEL: any = {
+  high: 'Severe',
+  moderate: 'Moderate',
+  low: 'Minor',
+  unknown: 'Unknown',
+};
+
+export const InteractionAlertsDialog: React.FC<Props> = (props) => {
+  const [interactions, setInteractions] = useState<MedicationInteractions>(structuredClone(props.interactions));
+
+  const allReasonsValid: boolean = [
+    ...(interactions.drugInteractions ?? []),
+    ...(interactions.allergyInteractions ?? []),
+  ].reduce((acc, val) => {
+    return acc && (val.overrideReason ?? '').trim().length > 0;
+  }, true);
+
+  const interactionTypeTitle = (title: string): ReactElement => {
+    return (
+      <Stack direction="row" spacing="12px" alignItems="center" style={{ marginTop: '16px' }}>
+        <Box
+          style={{
+            width: '19px',
+            height: '19px',
+            background: otherColors.priorityHighIcon,
+            borderRadius: '4px',
+            padding: '1px 2px 1px 2px',
+          }}
+        >
+          <PriorityHighOutlinedIcon style={{ width: '15px', height: '15px', color: '#FFF' }} />
+        </Box>
+        <Typography style={{ fontSize: '18px', fontWeight: 600, color: otherColors.priorityHighIcon }}>
+          {title}
+        </Typography>
+      </Stack>
+    );
+  };
+
+  const interactionSubtitleBox = (text: string): ReactElement => {
+    return (
+      <Box
+        style={{ background: otherColors.lightErrorBg, marginTop: '16px', padding: '16px', borderRadius: '4px' }}
+        display="flex"
+        alignItems="center"
+      >
+        <Typography variant="body2" style={{ color: otherColors.lightErrorText }}>
+          {text}
+        </Typography>
+      </Box>
+    );
+  };
+
+  const overrideReasonDropdown = (
+    value: string | undefined,
+    id: string,
+    onChange: (newValue: string) => void
+  ): ReactElement => {
+    if (value != null && !OVERRIDE_REASON.includes(value)) {
+      value = OTHER;
+    }
+    return (
+      <FormControl fullWidth sx={{ backgroundColor: 'white' }} size="small">
+        <InputLabel id={id}>Override reason</InputLabel>
+        <Select
+          label="Override reason"
+          labelId={id}
+          variant="outlined"
+          value={value ?? ''}
+          onChange={(e) => onChange(e.target.value)}
+        >
+          {OVERRIDE_REASON.map((reason) => {
+            return (
+              <MenuItem key={reason} value={reason}>
+                <Typography color="textPrimary" sx={{ fontSize: '16px' }}>
+                  {reason}
+                </Typography>
+              </MenuItem>
+            );
+          })}
+        </Select>
+      </FormControl>
+    );
+  };
+
+  const otherTextInput = (value: string | undefined, onChange: (newValue: string) => void): ReactElement => {
+    if (value == null || OVERRIDE_REASON.includes(value)) {
+      return <></>;
```

Comment:
`undefined` is used in other functions for this purposes here, I think it should be unified too
```suggestion
      return undefined;
```